### ISSUE TITLE
gcal ipython tutorial an FixedPoint

### ISSUE THREAD

**Issue by sarahcattan** [ioam/topographica#632]
I download the git version on 20 september.
I tried to run the gcal model with topographica -i and topographica -n.
In both, when i load the model i get:
WARNING:root:Time: 000000.00 main: Casting type 'FixedPoint' to Fraction.fraction
In the notebook gcal collector tutorial it came when entering:
topo.sim.model = ModelGCAL(cortex_density=47)
in command line it came after:
runscript(model_name)
I also tried to display figures after running 10000 simulations (using topographica in command line):
data = c(times=[10000])
The traceback go through holoviews/interface/collector.py and holoviews/core/ndmapping.py to finish in misc/fixedpoint.py with:
can't convert to FixedPoint: None
I have no idea where theses None values come from and what the Fraction.fraction in the first warning means.

**Comment by sarahcattan** [comment]
I solved it, it was not due to topogaphica.
If you are working with debain wheezy, this file
/lib/x86_64-linux-gnu/libc.so.6
need to be updated to import gmpy in python (FixedPoint was used because gmpy was not found when starting topograhica).
So i add the experimental depot in my source.list and i did
apt-get -t experimental install libc6-dev
to install the new version.
see here the complete description http://www.random-dev.de/fix-version-glibc_2-14-found-error/

**Comment by philippjfr** [comment]
Glad to hear you managed to resolve it. We should probably either fix the support for FixedPoint as a fallbackback for gmpy or deprecate it.

**Comment by mjabri** [comment]
Sarah, 
Are you running GCAL_Tutorial? Can you actually run this statement in the tutorial ok?
topo.sim.Retina.input_generator.anim(50)

**Comment by sarahcattan** [comment]
No, it give
Out[10]:
KeyError
None
And it appends each time i want an animation (more figures instead of the last time only)

**Comment by mjabri** [comment]
I see, so you get some patterns showing up. I don't get to see anything!! It seems the problem is not the patterngenerator anim() code but in the rendering of the dict returned by anim(). The dict seems ok as i can display the entries one by one.

**Comment by philippjfr** [comment]
I'll look at this now, will get back to you.

**Comment by mjabri** [comment]
Sarah, Unrelated but I am curious if you have tried the GUI with the examples/gcal.ty tutorial (ie topographica -g gcal.ty) and get to visualise projections after 1 iteration?

**Comment by sarahcattan** [comment]
No, one time i got it but it was VERY slow.

**Comment by mjabri** [comment]
The first iteration is somewhat slow for me too. I think it produces all the input patterns.

**Comment by sarahcattan** [comment]
If you want to run the GCAL model, i recommend you to use the file of the stable version of topographica, it run perfectly.

**Comment by philippjfr** [comment]
>The first iteration is somewhat slow for me too. I think it produces all the input patterns.

Topographica now initializes the model only when you run the first step, so that's what's taking so long.

**Comment by mjabri** [comment]
Ok, Thanks. Does the stable version has the sparse/GPU code in it?

**Comment by philippjfr** [comment]
>Does the stable version has the sparse/GPU code in it?

No, it doesn't unfortunately. Sorry I haven't managed to look at the GUI yet, hopefully I'll have some time in the next few days.

**Comment by mjabri** [comment]
That's great. Thanks.
